MaterialProgressBar/src/main/ets/components/MaterialProgressBar.ets

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2022 Application Library Engineering Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MaterialComponent from './basic/MaterialComponents'
+import MaterialProgressStyle from './materialprogressstyle'
+
+@Component
+struct MaterialProgressBar {
+  @State mStyle: MaterialProgressStyle = MaterialProgressStyle.MaterialCircular
+  modelBuild: any
+
+  aboutToAppear() {
+    console.log('style:' + this.mStyle)
+    console.log('modelBuild:' + JSON.stringify(this.modelBuild))
+    this.modelBuild.style = this.mStyle
+  }
+  onPageShow() {}
+
+  build() {
+    if (this.modelBuild) {
+      if (this.mStyle == MaterialProgressStyle.MaterialCircular) {
+        MaterialComponent(this.modelBuild)
+      } else if (this.mStyle == MaterialProgressStyle.MaterialHorizontal) {
+        MaterialComponent(this.modelBuild)
+      } else {
+        Text('undefined').fontSize(30)
+      }
+    }
+  }
+}
+
+namespace MaterialProgressBar {
+  export class MaterialModel {
+    isReversed: boolean = false
+    isMirror: boolean = false
+    isGradients: boolean = false
+    isStop: boolean = false
+    progress: number = 30
+    angle: number = 10
+    style: MaterialProgressStyle = MaterialProgressStyle.MaterialHorizontal
+    indeterminateAngle: boolean = false
+    indeterminate: boolean = false
+    radius: number = 50
+    color: Color = 0x029789
+    secondColor: Color = null
+    secondProgress: number = 0
+    bgColor: Color = null
+    strokeWidth: number = 8
```

Comment:
mark all dimentions as vp and font sizes as fp

Reply:
The model attributes have been defined as numbers, can't change there, done wherever possible